src/test/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMockTest.java

```diff
@@ -179,4 +180,91 @@ void runTaskTimerAsynchronously_TaskExecutedOnSeperateThread() throws Interrupte
 		assertFalse(scheduler.isQueued(testTask.getTaskId()));
 		assertEquals(2, count.get());
 	}
+
+	@Test
+	public void getPendingTasks_Sync()
+	{
+		assertEquals(0, scheduler.getPendingTasks().size());
+		int amountTasks = 20;
+		AtomicInteger count = new AtomicInteger(amountTasks);
+		Runnable callback = count::decrementAndGet;
+		for (int i = 0; i < amountTasks; i++)
+		{
+			scheduler.runTaskLater(null, callback, 2L+(i%5));
+		}
+		while (count.get()>0)
+		{
+			assertEquals(count.get(), scheduler.getPendingTasks().size());
+			scheduler.performOneTick();
+		}
+		assertEquals(count.get(), scheduler.getPendingTasks().size());
+	}
+
+	@Test
+	public void getPendingTasks_Async()
+	{
+		assertEquals(0, scheduler.getPendingTasks().size());
+		int amountTasks = 20;
+		AtomicInteger count = new AtomicInteger(amountTasks);
+		Runnable callback = () ->
+		{
+			try
+			{	// simulate some varying work load / execution time
+				Thread.sleep(ThreadLocalRandom.current().nextInt(2, 20));
+			}
+			catch (InterruptedException e)
+			{
+				e.printStackTrace();
+			}
+			count.decrementAndGet();
+		};
```

Comment:
Perhaps it would make this a bit more readable to just have a `private void simulateWorkload()` method in this class which can be called. This would get rid of all these inline lambda expressions.

`count` could be an optional parameter here.